### PR TITLE
Clarify that symbolic guarantees go into effect from 1.0.0

### PR DESCRIPTION
**Pull request by tigrannajaryan**
We merged the https://github.com/open-telemetry/opentelemetry-proto/pull/432 without clarifying when exactly the guarantees start applying. The intent was that the guarantees are needed for 1.0 release but we didn't make this explicit.

We are now making this explicit and until 1.0 released we should be free to fix the remaining known issues listed here:
https://github.com/open-telemetry/opentelemetry-proto/issues/456

Once 1.0 release is made we can remove the [from 1.0.0] labels from the README.md.

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]

**Review by arminru — APPROVED** [review]

**Review by carlosalberto — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
@open-telemetry/specs-approvers this has enough approvals. I will merge tomorrow if I do not see objections.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**
